tests/functional/s/super/superfluous_parens_walrus_py38.py

```diff
@@ -0,0 +1,86 @@
+"""Test the superfluous-parens warning with python 3.8 functionality (walrus operator)"""
+# pylint: disable=missing-function-docstring,invalid-name
+if not (x := False):
+    print(x)
+
+if not ((x := 1)):  # [superfluous-parens]
+    pass
+
+if not ((((x := 1)))):  # [superfluous-parens]
+    pass
+
+i = 1
+y = 1
+
+if odd := is_odd(i): # [undefined-variable]
+    pass
+if not (foo := 5):
+    pass
+
+if not ((x := y)):  # [superfluous-parens]
+    pass
+
+if ((x := y)):   # [superfluous-parens]
+    pass
+
+assert (ret := str(42))
+
+# False positive (str, tuple)
+A = [int(n) for n in (nums := '123')]
+B = [int(n) for n in (nums := ('1', '2', '3'))]
+
+# No false positive (list, set, dict)
+C = [int(n) for n in (nums := ['1', '2', '3'])]
+D = [int(n) for n in (nums := {'1', '2', '3'})]
+E = [int(n) for n in (nums := {'1': 1, '2': 2, '3': 3})]
+
+# False positive (brackets only)
+F = [int(n) for n in (nums := ['1'])]
+G = [int(n) for n in (nums := {'1'})]
+H = [int(n) for n in (nums := {'1': 1})]
+I = [int(n) for n in (nums := ('1'))]
+J = [int(n) for n in (nums := ('1', ))]
+
+# No false positive (brackets plus comma)
+K = [int(n) for n in (nums := ['1', ])]
+L = [int(n) for n in (nums := {'1', })]
+M = [int(n) for n in (nums := {'1': 1, })]
+
+# False positives (comma outside brackets)
+N = [int(next(iter(n))) for n in (nums := (['1'], ['2'], ['3']))]
+O = [int(next(iter(n))) for n in (nums := ({'1'}, {'2'}, {'3'}))]
+P = [int(next(iter(n))) for n in (nums := ({'1': 1}, {'2': 2}, {'3': 3}))]
+
+# False positives (aliasing)
+NUMS_LIST = ['1', '2', '3']
+NUMS_SET = {'1', '2', '3'}
+NUMS_DICT = {'1': 1, '2': 2, '3': 3}
+Q = [int(n) for n in (nums := NUMS_LIST)]
+R = [int(n) for n in (nums := NUMS_SET)]
+S = [int(n) for n in (nums := NUMS_DICT)]
+T = [int(n) for n in (nums := [*NUMS_LIST])]
+U = [int(n) for n in (nums := NUMS_LIST[:])]
+
+# 2D arrays with numpy (since they allow commas within indexing)
+import numpy
+ARRAY_2D = numpy.zeroes((3, 3), dtype=bool)
+V = not (vals := ARRAY_2D[:, :])  # No false positive
+X = not (vals := ARRAY_2D[2, 2])  # No false positive
+W = not (vals := ARRAY_2D[::])  # False positive
+Y = not (vals := ARRAY_2D[2:2])  # False positive
+Z = not (just_want_to_finish_the_alphabet := True)  # False positive
+
+x = ("Version " + __version__) in output
```

Comment:
I don't really understand these tests. What is output supposed to be? A string? A list? And where is __version__ taken from.

Reply:
It's one of the example a use case in 3249 (https://github.com/PyCQA/pylint/issues/3249#issuecomment-683424549). `__version__` was probably a string or this would fail. I think the parenthesis is really superfluous here

```suggestion
x = ("Version " + __version__) in output # [superfluous-parenthesis]
```